src/@shared/useCheckboxStore.tsx

```diff
@@ -19,15 +26,16 @@ const useCheckboxStore = create<CheckboxStore>((set, get) => ({
         values: state.isAllChecked ? new Set() : new Set(allValues),
         isAllChecked: !state.isAllChecked,
       })),
-    setValue: value => {
+    setCheckboxValue: ({ value, checkboxCount }) => {
       if (get().values.has(value)) {
         const newValues = new Set(get().values);
         newValues.delete(value);
 
-        set({ values: newValues });
+        set({ values: newValues, isAllChecked: false });
       } else {
         set(state => ({
           values: new Set(state.values).add(value),
+          isAllChecked: checkboxCount === state.values.size + 1 ? true : false,
```

Comment:
```suggestion
          isAllChecked: checkboxCount === (state.values.size + 1),
```
삼항 연산자로 분기해줄 필요는 없을 것 같습니다~